### ISSUE TITLE
Wrong number of tests reported in Flutter

### ISSUE THREAD

**Issue by tiagoferreira29** [phoenix-actions/test-reporting#51]
Hello,

I am using your test-report tool to showcase the test results regarding my flutter app, and i have discovered some discrepancies between counting of total tests and the actual file tests.

by using a simple json file generated by 

` flutter test --file-reporter json:"test-report.json"`

and getting the following json file:

```
{"protocolVersion":"0.1.1","runnerVersion":"1.24.9","pid":9500,"type":"start","time":0}
{"suite":{"id":0,"platform":"vm","path":""},"type":"suite","time":0}
{"test":{"id":1,"name":"loading repo","suiteID":0,"groupIDs":[],"metadata":{"skip":false,"skipReason":null},"line":null,"column":null,"url":null},"type":"testStart","time":0}
{"count":1,"time":4,"type":"allSuites"}
{"testID":1,"result":"success","skipped":false,"hidden":true,"type":"testDone","time":108779}
{"group":{"id":2,"suiteID":0,"parentID":null,"name":"","metadata":{"skip":false,"skipReason":null},"testCount":12,"line":null,"column":null,"url":null},"type":"group","time":108781}
{"test":{"id":3,"name":"(setUpAll)","suiteID":0,"groupIDs":[2],"hidden": true, "metadata":{"skip":false,"skipReason":null},"line":56,"column":3,"url":""},"type":"testStart","time":108781}
{"testID":3,"result":"success","skipped":false,"hidden":true,"type":"testDone","time":108805}
{"group":{"id":4,"suiteID":0,"parentID":2,"name":"[NFC Workflow] Multiple screen sizes","metadata":{"skip":false,"skipReason":null},"testCount":6,"line":119,"column":3,"url":""},"type":"group","time":108805}
{"test":{"id":5,"name":"Sample success test","suiteID":0,"groupIDs":[2,4],"metadata":{"skip":false,"skipReason":null},"line":165,"column":5,"url":"package:flutter_test/src/widget_tester.dart","root_line":120,"root_column":5,"root_url":""},"type":"testStart","time":108806}
{"testID":5,"result":"success","skipped":false,"hidden":false,"type":"testDone","time":120992}
{"test":{"id":6,"name":"(tearDownAll)","suiteID":0,"groupIDs":[2],"metadata":{"skip":false,"skipReason":null},"line":42,"column":5,"url":"package:integration_test/integration_test.dart"},"type":"testStart","time":210072}
{"testID":6,"result":"success","skipped":false,"hidden":true,"type":"testDone","time":210074}
{"success":true,"type":"done","time":210081}
```

It is expected to see one of two scenarios:

1. if the tool doesn't take into consideration the `"hidden":true` for the `(setupAll)` and `(tearDownAll)` the test reporting counts the tests as  `3 tests were completed in 10s, 3 passed, 0 failed and 0 skipped`

2.  if the tool doe take into consideration the `"hidden":true` for the `(setupAll)` and `(tearDownAll)` the test reporting counts the tests as  `1 tests were completed in 10s, 1 passed, 0 failed and 0 skipped`

The actual result is this one 
![Screenshot 2024-04-10 at 13 35 35](https://github.com/phoenix-actions/test-reporting/assets/164031554/2fe61fd4-a63e-4793-a2f2-c324834fb071)

So the counting is confusing

**Comment by dominicmh** [comment]
According to the documentation of the [TestDoneEvent](https://github.com/dart-lang/test/blob/master/pkgs/test/doc/json_reporter.md#testdoneevent), the hidden "tests" should be ignored:
> The `hidden` attribute indicates that the test's result should be hidden and not
counted towards the total number of tests run for the suite. This is true for
virtual tests created for loading test suites, `setUpAll()`, and
`tearDownAll()`. Only successful tests will be hidden.

This could be easily fixed by adapting [dart-json-parser.ts#L117-L118](https://github.com/phoenix-actions/test-reporting/blob/main/src/parsers/dart-json/dart-json-parser.ts#L117-L118) to:
```ts
      } else if (isTestDoneEvent(evt) && tests[evt.testID]) {
        if (!evt.hidden) {
          tests[evt.testID].testDone = evt
        } else {
          // remove hidden "test" from test report
          const indexOfHiddenTest = tests.indexOf(evt.testID, 0);
          if (indexOfHiddenTest > -1) {
            tests.splice(indexOfHiddenTest, 1);
          }
        }
```
@IanMoroney, would this solution be accepted as a PR? Let me know if you need more background information!

**Comment by IanMoroney** [comment]
Yep! More than happy to accept as a pr.
Let me know once done and I'll review and publish 👍

**Comment by dominicmh** [comment]
Great! The solution was slightly different in the end. It's fixed in #52

**Comment by IanMoroney** [comment]
Fixed by #55 
Publishing a new version now

**Comment by IanMoroney** [comment]
Fixed in https://github.com/phoenix-actions/test-reporting/releases/tag/v15
